include/accessor.h

```diff
@@ -51,6 +56,10 @@ class host_memory_layout {
 	 * - A two-dimensional layout is modelled as an interval of rows, which manifests as an offset (a multiple of the row width) and a stride (the row width
 	 * itself). Each row (aka 1-dimensional hyperplane) is modelled by the same one-dimensional layout.
 	 * - and so on for arbitrary dimensioned layouts.
+	 *
+	 *
+	 * Since contiguous dimensions can be merged when generating the memory layout, host_memory_layout is not generic over a fixed dimension count.
+	 * TODO: Implement merging of contiguous dimensions
 	 */
```

Comment:
We should improve the naming here. `dimension` has never really been accurate since we want to decouple buffer dimensionality from the number of strides by merging contiguous dimensions in the layout. Maybe `class stride` is a better option?

Reply:
I wasn't happy with having a breaking API change in the first place so I've decided to revert this to its original form, and I propose we move this discussion to a dedicated issue / PR.